### PR TITLE
More Ion 1.1 test cases

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

None

**Description of changes:**

I've made more changes to support Ion 1.1. They are broken into commits to make it easier to review. The commits are:

* Fix more IVMs in Ion 1.1 tests
* Removes VarInt and VarUInt tests that are inapplicable for Ion 1.1
* Adds more 'good' tests for macros

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: iontestdata_1_1/good/macros/void_invoked_in_list.ion [hunk 1]
```diff
@@ -0,0 +1,5 @@
+$ion_1_1
+[ (:void) ]
+[ (:void), ]
+[ a, (:void) ]
+[ (:void), b ]
```
Comment (zslayton) [inline iontestdata_1_1/good/macros/void_invoked_in_list.ion:5]: Perhaps one with multiple `(:void)`s?

File: iontestdata_1_1/good/macros/void_invoked_in_sexp.ion [hunk 2]
```diff
@@ -0,0 +1,4 @@
+$ion_1_1
+( (:void) )
+( a (:void) )
+( (:void) b )
```
Comment (zslayton) [inline iontestdata_1_1/good/macros/void_invoked_in_sexp.ion:4]: Same thought here.

**Review by zslayton — APPROVED** [review]

File: iontestdata_1_1/good/macros/void_invoked_in_sexp.ion (hunk 2 above)
Comment (popematt) [reply iontestdata_1_1/good/macros/void_invoked_in_sexp.ion:4]:
```suggestion
( a (:void) )
( (:void) b )
( (:void) (:void) c )
```

File: iontestdata_1_1/good/macros/void_invoked_in_list.ion (hunk 1 above)
Comment (popematt) [reply iontestdata_1_1/good/macros/void_invoked_in_list.ion:5]:
```suggestion
[ a, (:void) ]
[ (:void), b ]
[ (:void), (:void), c ]
```

**Pull request merged**